src/meta-srv/src/region/supervisor.rs

```diff
@@ -360,10 +360,9 @@ impl RegionSupervisor {
             Ok(false) => {}
             Ok(true) => {
                 warn!(
-                    "Skipping failover since maintenance mode is enabled. Removing failure detectors for regions: {:?}",
+                    "Skipping failover since maintenance mode is enabled. Detected region failures: {:?}",
                     regions
                 );
-                self.deregister_failure_detectors(regions).await;
```

Comment:
When the operator correctly sets maintenance mode, the current code that actively removes the failover detector causes the datanode to not trigger failover during scaling down. However, without the active removal of the failover detector logic, the failover detector would naturally stop triggering failure detection after receiving datanode heartbeats.